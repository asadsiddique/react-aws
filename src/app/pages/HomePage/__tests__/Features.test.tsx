import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Features } from '../Features';

const shallowRenderer = createRenderer();

describe('<Features />', () => {
  it.only('should render', () => {
    shallowRenderer.render(<Features />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
