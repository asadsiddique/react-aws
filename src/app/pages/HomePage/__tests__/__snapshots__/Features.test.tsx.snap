// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Features /> should render 1`] = `
<React.Fragment>
  <Title
    as="h2"
  >
    Features
  </Title>
  <Lead>
    Crafted for 
    <strong>
      highly scalable
    </strong>
    ,
     
    <strong>
      easily maintainable
    </strong>
     and
     
    <strong>
      highly performant
    </strong>
     React.js applications with a focus on
     
    <strong>
      best DX
    </strong>
     and 
    <strong>
      best practices
    </strong>
    .
  </Lead>
  <Features__List>
    <Features__Feature>
      <ForwardRef(SvgState)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          Predictable State
        </SubTitle>
        <P>
          Build easy to test and debug, flexible and extensible applications using
           
          <A
            href="https://redux.js.org/basics/usage-with-react"
            rel="noopener noreferrer"
            target="_blank"
          >
            Redux
          </A>
          . Unidirectional data flow allows for change logging and time travel debugging.
           
          <A
            href="https://github.com/zalmoxisus/redux-devtools-extension"
            rel="noopener noreferrer"
            target="_blank"
          >
            Install Chrome Redux Dev Tools
          </A>
           
          to see how your application's state changes and travel in time to debug. Type any github username below and see it in action with Redux Dev Tools.
        </P>
        <GithubRepoForm />
      </Features__Content>
    </Features__Feature>
    <Features__Feature>
      <ForwardRef(SvgCss)
        className="feature-icon"
      />
      <Features__Content>
        <SubTitle>
          Next Generation CSS
        </SubTitle>
        <P>
          Write composable CSS thatâ€™s co-located with your components for complete modularity. Ship only the styles that are on the page for the best performance. Generate application-wide styles and themes for your components. Change the theme below to see how easy and intuitive theming can ever be!
        </P>
        <ThemeSwitch />
      </Features__Content>
    </Features__Feature>
    <Link
      to="/notfound"
    >
      /NotFound
    </Link>
  </Features__List>
</React.Fragment>
`;
